core/Azure.Mcp.Core/src/Areas/Server/Models/ConsolidatedToolDefinition.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json.Serialization;
+using Azure.Mcp.Core.Commands;
+
+namespace Azure.Mcp.Core.Areas.Server.Models;
+
+/// <summary>
+/// Represents a composite tool definition that groups multiple related Azure operations together.
+/// Used to create consolidated tools from the azure_mcp_consolidated_tools JSON configuration.
+/// </summary>
+public sealed class ConsolidatedToolDefinition
+{
+    /// <summary>
+    /// Gets or sets the name of the composite tool.
+    /// </summary>
+    [JsonPropertyName("name")]
+    public string Name { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the description of the composite tool's capabilities and purpose.
+    /// </summary>
+    [JsonPropertyName("description")]
+    public string Description { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the tool metadata containing capability information.
+    /// </summary>
+    [JsonPropertyName("toolMetadata")]
+    public ToolMetadata? ToolMetadata { get; set; }
+
+    /// <summary>
+    /// Gets or sets the list of tool names that are mapped to this consolidated tool.
+    /// </summary>
+    [JsonPropertyName("mappedToolList")]
+    public List<string> MappedToolList { get; set; } = new List<string>();
```

Comment:
Looking at the `consolidated-tools.json‎` file entries, it seems that all the properties of the ConsolidatedToolDefinition model are required. If that’s the case, can we mark them as `required` and use `init` (instead of `set`) and avoid nullable (`?`)?